### DISCUSSION TITLE
New Trigger Form: how to change the logical date, if the DAG has no params?

### DISCUSSION THREAD

**Discussion by wavewater** [apache/airflow#38869] [Q&A] [answered]
In Airflow 2.3.4 and other old versions, there was a very simple trigger form which received a major upgrade lately.

So in Airflow 2.3.4 if you manually triggered the DAG from the UI with the Right-Arrow ICON (up rigth corner), you in fact had a choice:

1. Trigger DAG
2. Trigger DAG w/config

With this, if you wanted, you could select "Trigger DAG w/config" and then set the Logical date using the date picker. The main point here: You could ALWAYS select "Trigger DAG w/config" even if the DAG had no Dag params defined at all and thus CHANGE the Logical Date!

Now in Airflow 2.8.3, if you want to trigger a Dag manually via the UI using the Right-Arrow ICON, if there is no Dag params defined in the Dag, the Trigger Form does not open at all, so via UI you simply cannot run a Dag with a different logical Date. 

Could it be that by introducing the New Trigger Form accidently the capability was lost to set the logical date for DAGs that have no dag params defined? If yes, is this a bug? 

Is there a way, to always open the Trigger Form? In that case, actually one can set the logical date using the date picker in the section "Generated Configuration JSON and Dagrun Option".

**Comment by Taragolis** [comment, marked as answer]
DagRun trigger for for non parametrised DAGs is disabled by default since Airflow 2.7, for enable it you need set [`[webserver] show_trigger_form_if_no_params`](https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#show-trigger-form-if-no-params) configuration to `True`

Additional links for the reference:
- [Airflow 2.7.0 Release Notes](https://airflow.apache.org/docs/apache-airflow/stable/release_notes.html#the-trigger-ui-form-is-skipped-in-web-ui-if-no-parameters-are-defined-in-a-dag-33351)
- [AIP-50 Trigger DAG UI Extension with Flexible User Form Concept](https://cwiki.apache.org/confluence/display/AIRFLOW/AIP-50+Trigger+DAG+UI+Extension+with+Flexible+User+Form+Concept)

**Reply by wavewater** [reply]
Thank you, it worked as you stated. We added AIRFLOW_WEBSERVER_SHOW_TRIGGER_FORM_IF_NO_PARAMS: "true" in the docker compose file, restarted Airflow WebUI service and then "Trigger Dag w/config" is available even if there are no Dag params in the dag. Marking yours as answer.
